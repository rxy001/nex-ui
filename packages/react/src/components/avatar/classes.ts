import { generateUtilityClasses } from '../utils'

export const avatarClasses = generateUtilityClasses('nui-avatar', [
  'root',

  'radius-sm',
  'radius-md',
  'radius-lg',
  'radius-xl',
  'radius-full',
  'radius-none',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'outlined',

  'img',
])

export const avatarGroupClasses = generateUtilityClasses('nui-avatar-group', [
  'root',
  'surplus',
])
