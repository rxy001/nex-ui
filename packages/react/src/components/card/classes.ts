import { generateUtilityClasses } from '../utils'

export const cardClasses = generateUtilityClasses('nui-card', [
  'root',
  'blurred',
  'hoverable',

  'shadow-xs',
  'shadow-sm',
  'shadow-md',
  'shadow-lg',
  'shadow-xl',

  'radius-none',
  'radius-sm',
  'radius-md',
  'radius-lg',
])

export const cardHeaderClasses = generateUtilityClasses('nui-card-header', [
  'root',
  'content',
  'title',
  'subtitle',
])

export const cardBodyClasses = generateUtilityClasses('nui-card-body', ['root'])

export const cardFooterClasses = generateUtilityClasses('nui-card-footer', [
  'root',
])

export const cardActionAreaClasses = generateUtilityClasses(
  'nui-card-action-area',
  ['root'],
)
