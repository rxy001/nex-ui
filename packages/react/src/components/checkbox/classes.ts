import { generateUtilityClasses } from '../utils'

// 用于单元测试
export const checkboxClasses = generateUtilityClasses('nui-checkbox', [
  // ---------root---------
  'root',

  // "sm" | "md" | "lg" | "full" | "none"
  'radius-sm',
  'radius-md',
  'radius-lg',
  'radius-full',
  'radius-none',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'checked',
  'disabled',
  'indeterminate',
  // ---------root---------

  'input',
  'label',
  'icon',
])

export const checkboxGroupClasses = generateUtilityClasses(
  'nui-checkbox-group',
  [
    'root',
    'label',
    'wrapper',
    'orientation-vertical',
    'orientation-horizontal',
  ],
)
