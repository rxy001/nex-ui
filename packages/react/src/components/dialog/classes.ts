import { generateUtilityClasses } from '../utils'

export const dialogClasses = generateUtilityClasses('nui-dialog', [
  'root',
  'open',
  'backdrop',
])

export const dialogContentClasses = generateUtilityClasses(
  'nui-dialog-content',
  [
    'root',

    'size-xs',
    'size-sm',
    'size-md',
    'size-lg',
    'size-xl',
    'size-full',

    'full-screen',

    'placement-top',
    'placement-bottom',
    'placement-center',

    'scroll-inside',
    'scroll-outside',

    'paper',
    'close-button',
  ],
)

export const dialogHeaderClasses = generateUtilityClasses('nui-dialog-header', [
  'root',
])

export const dialogFooterClasses = generateUtilityClasses('nui-dialog-footer', [
  'root',
])

export const dialogBodyClasses = generateUtilityClasses('nui-dialog-body', [
  'root',
])
