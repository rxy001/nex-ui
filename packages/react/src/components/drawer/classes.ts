import { generateUtilityClasses } from '../utils'

export const drawerClasses = generateUtilityClasses('nui-drawer', [
  'root',
  'open',
  'backdrop',
])

export const drawerContentClasses = generateUtilityClasses(
  'nui-drawer-content',
  [
    'root',

    'placement-top',
    'placement-right',
    'placement-bottom',
    'placement-left',

    'size-xs',
    'size-sm',
    'size-md',
    'size-lg',
    'size-xl',
    'size-full',

    'paper',
    'close-button',
  ],
)

export const drawerHeaderClasses = generateUtilityClasses('nui-drawer-header', [
  'root',
])

export const drawerFooterClasses = generateUtilityClasses('nui-drawer-footer', [
  'root',
])

export const drawerBodyClasses = generateUtilityClasses('nui-drawer-body', [
  'root',
])
