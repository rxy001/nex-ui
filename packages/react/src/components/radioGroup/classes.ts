import { generateUtilityClasses } from '../utils'

// 用于单元测试
export const radioClasses = generateUtilityClasses('nui-radio', [
  // ---------root---------
  'root',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'checked',
  'disabled',
  // ---------root---------

  'input',
  'label',
  'dot',
])

export const radioGroupClasses = generateUtilityClasses('nui-radio-group', [
  'root',
  'label',
  'wrapper',
  'orientation-vertical',
  'orientation-horizontal',
])
