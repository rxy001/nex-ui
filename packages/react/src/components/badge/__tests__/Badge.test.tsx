import {
  testClassNamesForwarding,
  testColorDataAttrs,
  testComponentStability,
  testRefForwarding,
  testRootClassName,
  testSlotPropsForwarding,
  testVariantDataAttrs,
  renderWithNexUIProvider,
} from '~/tests/shared'
import { Badge } from '../index'
import { badgeSlotClasses } from './classes'

const slots = ['closeButton'] as const

describe('Badge', () => {
  testComponentStability(<Badge />)

  testRootClassName(<Badge />)

  testRefForwarding(<Badge />)

  testColorDataAttrs(<Badge />)

  testVariantDataAttrs(<Badge />, ['size', ['sm', 'md', 'lg', 'xl']])

  testVariantDataAttrs(<Badge />, [
    'radius',
    ['none', 'sm', 'md', 'lg', 'xl', 'full'],
  ])

  testClassNamesForwarding(
    <Badge closable />,
    slots,
    {
      closeButton: 'test-close-button',
    },
    badgeSlotClasses,
  )

  testSlotPropsForwarding(
    <Badge closable />,
    slots,
    {
      closeButton: { className: 'test-close-button' },
    },
    badgeSlotClasses,
  )

  it('should render with default props', () => {
    const { container } = renderWithNexUIProvider(<Badge>Badge</Badge>)
    const badgeRoot = container.firstElementChild
    expect(badgeRoot).toHaveClass(badgeSlotClasses.root)
    expect(badgeRoot).toHaveAttribute('data-size', 'md')
    expect(badgeRoot).toHaveAttribute('data-radius', 'md')
    expect(badgeRoot).toHaveAttribute('data-variant', 'solid')
    expect(badgeRoot).toHaveAttribute('data-color', 'blue')
    expect(badgeRoot).toHaveAttribute('data-closable', 'false')
    expect(badgeRoot).toHaveAttribute('data-disabled', 'false')

    expect(badgeRoot).toMatchSnapshot()
  })

  it('should render close button when closable is true', async () => {
    const handleClose = jest.fn()

    const { getByLabelText, user } = renderWithNexUIProvider(
      <Badge closable onClose={handleClose}>
        Badge
      </Badge>,
    )
    const closeButton = getByLabelText('Close Badge')
    expect(closeButton).toBeInTheDocument()

    await user.click(closeButton)

    expect(handleClose).toHaveBeenCalledTimes(1)
  })

  it('should disable badge when disabled is true', () => {
    const { container } = renderWithNexUIProvider(<Badge disabled>Badge</Badge>)
    const badgeRoot = container.firstElementChild
    expect(badgeRoot).toHaveAttribute('data-disabled', 'true')

    expect(badgeRoot).toHaveStyleRule('pointer-events', 'none')
  })
})
