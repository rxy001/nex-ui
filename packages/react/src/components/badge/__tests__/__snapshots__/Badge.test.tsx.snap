// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render with default props 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  gap: var(--nui-spaces-1);
  line-height: 1rem;
  font-size: var(--nui-font-sizes-md);
  -webkit-padding-start: var(--nui-spaces-2-5);
  padding-inline-start: var(--nui-spaces-2-5);
  -webkit-padding-end: var(--nui-spaces-2-5);
  padding-inline-end: var(--nui-spaces-2-5);
  height: var(--nui-sizes-6);
  border-radius: var(--nui-radii-md);
  background-color: var(--nui-colors-blue-primary);
  color: var(--nui-colors-blue-contrast-text);
}

.emotion-0 svg {
  font-size: 1.25em;
}

<span
  class="nui-badge-root emotion-0"
  data-closable="false"
  data-color="blue"
  data-disabled="false"
  data-radius="md"
  data-size="md"
  data-variant="solid"
>
  Badge
</span>
`;
